### ISSUE TITLE
ERA001 false positive on examples in comments

### ISSUE THREAD

**Issue by 9999years** [astral-sh/ruff#10072]
Ruff treats this as commented-out code, despite being an example.

```
# NOTE: Only upper-case fields in `extra` will be sent to `journald`.
# For instance, this:
#
#     logging.error("Uh oh!", extra={"payload": "my-payload"})
#
# Will not show up as a `payload` field in `journalctl`, but this will:
#
#     logging.error("Uh oh!", extra={"PAYLOAD": "my-payload"})
journal_handler = JournalHandler()
```

It would be nice if Ruff could recognize "for example" or `` ```python ``-blocks as "example code".

**Comment by MichaReiser** [comment]
Hmm that's an interesting one. The rule is working as intended but I can see how it is undesired in this specific case. 

I would favor the solution where the rule ignores `Python` code blocks, although that would require that the rule supports some form of markdown parsing, which seems a bit much.

**Comment by omh1280** [comment]
It's a bit of a hack, but you can use triple quotes to get around this case https://play.ruff.rs/42481cc7-c07b-4158-9b6b-c65f0be7c09c

**Comment by kurellajunior** [comment]
Mine is different but matches the title. Maybe the rule is too eager?

```python
    # case 1: both codes are CREATE
    if cond1:
        pass
    # case 2: one code is a CREATE, the other is an existing 
    elif cond2:
        pass
    # case 3: both codes are existing
    elif cond3:
        pass
```

**Comment by rmorshea** [comment]
A reasonable heuristic might be to allow code in comments so long as there's a preceding explanation. In particular, this means the rule would check that commented code is immediately preceded by comment lines with non-code content.

Ok cases:

```python
# here's an example:
# x = y + x
```

```python
# here's an example:
#
# x = y + x
```

Bad cases:

```python
# x = y + x
```

```
#
# x = y + x
```

```
# here's an example:

# x = y + x
```

```
more_code = here  # here's an example:
# x = y + x
```

**Comment by rmorshea** [comment]
@kurellajunior I'd argue your issue is a bit different. You're using natural language but ruff is identifying it as code. This particular issue focuses on cases in which you actually want to include code in your comment because it serves as an example.

More specifically, `case N:` is likely being confused with the `match` statement syntax. If you change to use `condition N:` I suspect it won't match anymore. Granted, that still looks like an issue since what follows the `:` should be enough to ignore the line.

**Comment by kurellajunior** [comment]
> @kurellajunior I'd argue your issue is a bit different. You're using natural language but ruff is identifying it as code. This particular issue focuses on cases in which you actually want to include code in your comment because it serves as an example.
> 

Fully agree, it is a bit different. I thought it might still fit here, if someone is looking into the rule, to make it more robust to keep this case in mind too. I am happy to open a separate issue if this would be prefered.

And yes, changing the wording to a synonym solves the issue (or removing the colon or else). So I considered it not pressing enough for a separate issue

**Comment by Danipulok** [comment]
I have the same issue:
```python
# simple comment1
# TASKS: processing
# simple comment2
```

`ruff check --select=ERA001 mre.py`:
```bash
$ ruff check --select=ERA001 mre.py 
mre.py:2:1: ERA001 Found commented-out code
  |
1 | # simple comment1
2 | # TASKS: processing
  | ^^^^^^^^^^^^^^^^^^^ ERA001
3 | # simple comment2
  |
  = help: Remove commented-out code

Found 1 error.
```
